src/pages/Guilds/Guilds.tsx

```diff
@@ -55,18 +53,27 @@ const GuildsPage: React.FC = () => {
         <Filter />
         {!error ? (
           <ProposalsList data-testid="proposals-list">
-            {!proposalsIds ? (
-              <>
-                <SkeletonProposalCard />
-                <SkeletonProposalCard />
-              </>
-            ) : (
+            {proposalsIds ? (
               proposalsIds.map(proposalId => (
-                <ProposalCard
-                  id={proposalId}
-                  href={`/${chainName}/${guildId}/proposal/${proposalId}`}
-                />
+                <InView>
+                  {({ inView, ref }) => (
+                    <div ref={ref}>
+                      <ProposalCard
+                        id={inView ? proposalId : null}
+                        href={`/${chainName}/${guildId}/proposal/${proposalId}`}
+                      />
+                    </div>
+                  )}
+                </InView>
               ))
+            ) : (
+              <>
+                <ProposalCard />
+                <ProposalCard />
+                <ProposalCard />
+                <ProposalCard />
+                <ProposalCard />

```

Comment:
Is there a better way to have a default loading state for when proposalIds are loading?. I'm not entirely sure what best practises here would be.

Reply:
I'm not sure if there's a significantly better way, as we somehow need to render an array of cards here...